clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -2014,6 +2238,370 @@ test_synthetic_time_quanta()
+                    assert(false);
+                }
+                if (type_is_instr(memref.instr.type)) {
+                    sched += ('A' + (memref.instr.tid - TID_BASE) % 26);
+                    if (expect_pc_or_marker_val != 0 &&
+                        memref.instr.addr != expect_pc_or_marker_val) {
+                        std::cerr << "Expected pc " << expect_pc_or_marker_val
+                                  << " != " << memref.instr.addr << " at time " << time
+                                  << "\n";
+                        assert(false);
+                    }
+                    if (expect_indirect_branch_target != 0 &&
+                        memref.instr.indirect_branch_target !=
+                            expect_indirect_branch_target) {
+                        std::cerr << "Expected indirect_branch_target "
+                                  << expect_indirect_branch_target
+                                  << " != " << memref.instr.indirect_branch_target
+                                  << " at time " << time << "\n";
+                        assert(false);
+                    }
+                }
+                if (memref.marker.type == TRACE_TYPE_MARKER) {
+                    if (expect_marker_type != TRACE_MARKER_TYPE_RESERVED_END) {
+                        if (memref.marker.marker_type != expect_marker_type) {
+                            std::cerr << "Expected marker_type " << expect_marker_type
+                                      << " != " << memref.marker.marker_type
+                                      << " at time " << time << "\n";
+                            assert(false);
+                        }
+                        if (expect_pc_or_marker_val != 0 &&
+                            memref.marker.marker_value != expect_pc_or_marker_val) {
+                            std::cerr
+                                << "Expected marker_value " << expect_pc_or_marker_val
+                                << " != " << memref.marker.marker_value << " at time "
+                                << time << "\n";
+                            assert(false);
+                        }
+                    }
+                    switch (memref.marker.marker_type) {
+                    case TRACE_MARKER_TYPE_CONTEXT_SWITCH_START:
+                    case TRACE_MARKER_TYPE_CONTEXT_SWITCH_END: sched += "c"; break;
+                    case TRACE_MARKER_TYPE_SYSCALL_TRACE_START:
+                    case TRACE_MARKER_TYPE_SYSCALL_TRACE_END: sched += "s"; break;
+                    default: sched += "."; break;
+                    }
+                }
+            } else if (status != scheduler_t::STATUS_EOF) {
+                sched += "?";
+            }
+        };
+        static constexpr uint64_t instr_1_tid_c_pc = 100 * 2 + 10;
+        static constexpr uint64_t instr_2_tid_a_pc = 100 * 0 + 30;
+        static constexpr uint64_t instr_3_tid_a_pc = 100 * 0 + 50;
```

Comment:
Use named constants for all these magic numbers which presumably are shared with values up above but are currently duplicated here.